### PR TITLE
Add command catch KeyboardInterrupt and restore original_content (2933, 318)

### PR DESCRIPTION
**Pull request by estyxx**
# Pull Request Check List

Resolves: #2933 #318 

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

I wrapped KeyboardInterrupt where we are changing the content and write it... 
At line 103 I also write `original_content` instead of content, because if there isn't any package to add I will expect that nothing change... let me know if I'm wrong... 

FYI this issue is related to Hacktoberfest

### CODE REVIEW

File: poetry/console/commands/add.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def handle(self):
         packages = [name for name in packages if name not in existing_packages]
 
         if not packages:
-            self.poetry.file.write(content)
+            self.poetry.file.write(original_content)
```
Comment (abn) [inline poetry/console/commands/add.py:103]: Should we just remove this as we do not modify anything before this?

File: poetry/console/commands/add.py [hunk 2]
```diff
@@ -152,29 +152,28 @@ def handle(self):
...
+            if self.option("lock"):
+                self._installer.lock()
 
-        self._installer.whitelist([r["name"] for r in requirements])
+            self._installer.whitelist([r["name"] for r in requirements])
 
-        try:
             status = self._installer.run()
-        except Exception:
+        except BaseException:
```
Comment (abn) [inline poetry/console/commands/add.py:175]: Maybe add a comment here so we retain why we made this change.

**Review by abn — CHANGES_REQUESTED** [review]
Couple of minor comments.

File: poetry/console/commands/add.py (hunk 1 above)
Comment (estyxx) [reply poetry/console/commands/add.py]: yes, think so, tests seem to work anyway...

File: poetry/console/commands/add.py (hunk 2 above)
Comment (estyxx) [reply poetry/console/commands/add.py:174]: added!

File: poetry/console/commands/add.py [hunk 3]
```diff
@@ -173,6 +172,7 @@ def handle(self):
 
             status = self._installer.run()
         except BaseException:
+            # Catch all exception including KeyboardInterrupt in order to restore the original content
```
Comment (abn) [inline poetry/console/commands/add.py:175]:
```suggestion
            # Using BaseException here as some exceptions, eg: KeyboardInterrupt, do not inherit from Exception
```

Sorry, should have been more specific about what I had in mind.

**Review by abn — APPROVED** [review]

**Comment by abn** [comment]
@estyxx thank you for your contributions!

**Pull request merged**